### DISCUSSION TITLE
firefox: program is automatically started outside of firejail (KDE Plasma)

### DISCUSSION THREAD

**Discussion by OtherJohnGray** [netblue30/firejail#5244] [Q&A] [answered]
I installed firejail version 0.9.70 from Debian 11 backports, and ran firecfg. 

After starting firefox, firejail --list shows a firefox jail:

    12880:user::/usr/bin/firejail /usr/bin/firefox-esr

and ps -x --forrest shows a process tree rooted in /usr/bin/plasmashell and going through /usr/bin/firejail:

       4689 ?        Sl     0:09 /usr/bin/plasmashell
      12880 ?        S      0:00  \_ /usr/bin/firejail /usr/bin/firefox-esr
      12882 ?        Sl     0:00      \_ /usr/bin/xdg-dbus-proxy --fd=4 --args=5
      12884 ?        S      0:00      \_ /usr/bin/firejail /usr/bin/firefox-esr
      12894 ?        Sl     2:10          \_ /usr/bin/firefox-esr
      13055 ?        Sl     0:02              \_ /usr/lib/firefox-esr/firefox-esr -contentproc -childID 1 -isForBrowser -prefsLen 1 -prefMapSize 240017 -jsI
      13098 ?        Sl     0:02              \_ /usr/lib/firefox-esr/firefox-esr -contentproc -childID 2 -isForBrowser -prefsLen 96 -prefMapSize 240017 -js
      13122 ?        Sl     0:01              \_ /usr/lib/firefox-esr/firefox-esr -contentproc -childID 3 -isForBrowser -prefsLen 4967 -prefMapSize 240017 -
      13290 ?        Sl     1:16              \_ /usr/lib/firefox-esr/firefox-esr -contentproc -childID 4 -isForBrowser -prefsLen 5757 -prefMapSize 240017 -
      13549 ?        Sl     0:17              \_ /usr/lib/firefox-esr/firefox-esr -contentproc -childID 5 -isForBrowser -prefsLen 5804 -prefMapSize 240017 -
      13669 ?        Sl     0:00              \_ /usr/lib/firefox-esr/firefox-esr -contentproc -parentBuildID 20220623065118 -prefsLen 5804 -prefMapSize 240
      14308 ?        Sl     0:03              \_ /usr/lib/firefox-esr/firefox-esr -contentproc -childID 6 -isForBrowser -prefsLen 5804 -prefMapSize 240017 -
      15790 ?        Sl     0:19              \_ /usr/lib/firefox-esr/firefox-esr -contentproc -childID 7 -isForBrowser -prefsLen 5804 -prefMapSize 240017 -
      17169 ?        Sl     0:21              \_ /usr/lib/firefox-esr/firefox-esr -contentproc -childID 8 -isForBrowser -prefsLen 5804 -prefMapSize 240017 -
      20241 ?        Sl     0:00              \_ /usr/lib/firefox-esr/firefox-esr -contentproc -childID 9 -isForBrowser -prefsLen 5804 -prefMapSize 240017 -

However, if I then restart my machine, firefox automatically opens following restart, however now firejail --list shows no output, and ps -x --forest shows firefox running in a process tree rooted at /usr/bin/ksmserver and NOT going through firejail at all.

       4654 ?        Sl     0:03 /usr/bin/plasmashell
       4486 ?        Sl     0:00 /usr/bin/ksmserver
       4852 ?        Sl     0:00  \_ /usr/lib/x86_64-linux-gnu/libexec/DiscoverNotifier -session 10153152141df000165740923600000048910004_1657427968_209091
       5078 ?        Sl     0:00  \_ /usr/bin/konsole -session 10153152141df000165735250100000043700011_1657427968_209122
       5455 pts/2    Ss     0:00  |   \_ /bin/bash
       7163 pts/2    R+     0:00  |   |   \_ ps -x --forest
       5466 pts/1    Ss     0:00  |   \_ /bin/bash
       5330 ?        Sl     0:00  \_ /usr/bin/dolphin -session 10153152141df000165740900600000050340013_1657427968_209136
       5358 ?        Sl     0:06  \_ /usr/lib/firefox-esr/firefox-esr --sm-client-id 10153152141df000165742706400000043640014
       6425 ?        Sl     0:01      \_ /usr/lib/firefox-esr/firefox-esr -contentproc -childID 1 -isForBrowser -prefsLen 1 -prefMapSize 239943 -jsInit 2856
       6483 ?        Sl     0:00      \_ /usr/lib/firefox-esr/firefox-esr -contentproc -childID 2 -isForBrowser -prefsLen 5081 -prefMapSize 239943 -jsInit 2
       6572 ?        Sl     0:00      \_ /usr/lib/firefox-esr/firefox-esr -contentproc -childID 3 -isForBrowser -prefsLen 5741 -prefMapSize 239943 -jsInit 2
       6594 ?        Sl     0:00      \_ /usr/lib/firefox-esr/firefox-esr -contentproc -childID 4 -isForBrowser -prefsLen 5741 -prefMapSize 239943 -jsInit 2

Does this mean firefox is no longer running via firejail? This seems like quite a security trap, especially if one has firefox set to open previous tabs  on restart... or am I missing something?

**Comment by rusty-snake** [comment, marked as answer]
> Does this mean firefox is no longer running via firejail? 

Yes. Known, see #3539.

**Comment by OtherJohnGray** [comment]
Thanks - I tried searching but didn’t find that one. 

I’ve already implemented the “don’t restart previous session”
workaround.

On July 10, 2022, Richard Laager ***@***.***> wrote:
> > Does this mean firefox is no longer running via firejail?
> >
>
> Yes. Known, see #3539
> <https://github.com/netblue30/firejail/issues/3539>.
>

—
Reply to this email directly, view it on GitHub
<https://github.com/netblue30/firejail/discussions/5244#discussioncomment-
3114717>, or unsubscribe <https://github.com/notifications/unsubscribe-
auth/AQVVLGLADUWAL4HHCT7KWLTVTJ7VNANCNFSM53ENLG5Q>.
You are receiving this because you authored the thread.Message ID:
***@***.***>
